core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala

```diff
@@ -107,24 +110,26 @@ private[r2dbc] class PostgresDurableStateDao(executorProvider: R2dbcExecutorProv
     }
   }
 
-  protected def selectStateSql(slice: Int, entityType: String): String = {
-    val stateTable = settings.getDurableStateTableWithSchema(entityType, slice)
-    sql"""
-    SELECT revision, state_ser_id, state_ser_manifest, state_payload, db_timestamp
-    FROM $stateTable WHERE persistence_id = ?"""
-  }
+  protected def selectStateSql(slice: Int, entityType: String): String =
+    sqlCache.get(slice, s"selectStateSql-$entityType") {
```

Comment:
when reviewing this, the highest risk of mistake (by me) is using the wrong cache key, such as accidentally using the same in several places